azurelinuxagent/common/protocol/wire.py

```diff
@@ -1101,33 +1107,28 @@ def report_event(self, event_list):
                                          "Single event too large: {0}, with the length: {1} more than the limit({2})"
                                          .format(str(details_of_event), len(event_str), MAX_EVENT_BUFFER_SIZE))
                     continue
+
+                # If buffer is full, send out the events in buffer and reset buffer
                 if len(buf[event.providerId] + event_str) >= MAX_EVENT_BUFFER_SIZE:
-                    self.send_encoded_event(event.providerId, buf[event.providerId])
+                    logger.verbose("No of events this request = {0}".format(events_per_request[event.providerId]))
+                    _send_event(event.providerId, debug_info)
                     buf[event.providerId] = b''
-                    logger.verbose("No of events this request = {0}".format(events_per_request))
-                    events_per_request = 0
+                    events_per_request[event.providerId] = 0
+
+                # Add encoded events to the buffer
                 buf[event.providerId] = buf[event.providerId] + event_str
-                events_per_request += 1
-            except UnicodeError as e: # pylint: disable=C0103
-                unicode_error_count += 1
-                if len(unicode_errors) < max_send_errors_to_report:
-                    unicode_errors.append(ustr(e))
-            except Exception as e: # pylint: disable=C0103
-                event_report_error_count += 1
-                if len(event_report_errors) < max_send_errors_to_report:
-                    event_report_errors.append(ustr(e))
+                events_per_request[event.providerId] += 1
```

Comment:
why is buf initialized to b'' when we concatenate strings?

Reply:
This change was added when I made the changes for the unicode stuff. Basically we encode the xml string that we need to send out and then save it in the buffer. Since we keep adding the encoded string to the buffer, the buffer had to be initialized with an encoded string (`b""`)